devicetypes/heltun/heltun-ht01-thermostat.src/heltun-ht01-thermostat.groovy

```diff
@@ -0,0 +1,538 @@
+		map.name = "temperature"
+		map.value = scaledTemp
+		map.unit = locaScale
+		sendEvent(map)
+	} else if (floorTemperature == cmd.sensorType) {
+		def deviceTemp = cmd.scaledSensorValue
+		def scaledTemp = (deviceScale == locaScale) ? deviceTemp : (deviceScale == "F" ? roundC(fahrenheitToCelsius(deviceTemp)) : celsiusToFahrenheit(deviceTemp).toDouble().round(0).toInteger())
+		map.name = "temperature"
+		map.value = scaledTemp
+		map.unit = locaScale
+		child?.sendEvent(map)
+	} else if (himidity == cmd.sensorType) {
+		map.name = "humidity"
+		map.value = cmd.scaledSensorValue.toInteger()
+		map.unit = "%"
+		sendEvent(map)
+	} else if (illuminance == cmd.sensorType) {
+		map.name = "illuminance"
+		map.value = cmd.scaledSensorValue
+		sendEvent(map)
+	}	
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.meterv3.MeterReport cmd) {
+	def map = [:]
+	if (cmd.meterType == 1) {
+		if (cmd.scale == 0) {
+			map.name = "energy"
+			map.value = cmd.scaledMeterValue
+			map.unit = "kWh"
+		} else if (cmd.scale == 2) {
+			map.name = "power"
+			map.value = Math.round(cmd.scaledMeterValue)
+			map.unit = "W"
+		}else if (cmd.scale == 4) {
+			map.name = "voltage"
+			map.value = Math.round(cmd.scaledMeterValue)
+			map.unit = "V"
+		}
+		sendEvent(map)
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.thermostatoperatingstatev2.ThermostatOperatingStateReport cmd) {
+	def state = (cmd.operatingState == 1) ? "heating" : "idle"
+	sendEvent(name: "thermostatOperatingState", value: state)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.thermostatsetpointv2.ThermostatSetpointReport cmd) {
+	def locaScale = getTemperatureScale() //HubScale
+	def deviceScale = (cmd.scale == 1) ? "F" : "C" //DeviceScale
+	def deviceTemp = cmd.scaledValue
+	def setPoint = (deviceScale == locaScale) ? deviceTemp : (deviceScale == "F" ? roundC(fahrenheitToCelsius(deviceTemp)) : celsiusToFahrenheit(deviceTemp).toDouble().round(0).toInteger())
+	def mode = modetonumMap[device.currentValue("thermostatMode")]
+	if (mode == 0) {setPoint = 0}
+	sendEvent(name: "heatingSetpoint", value: setPoint, unit: locaScale)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.thermostatmodev2.ThermostatModeSupportedReport cmd) {
+	def tSupportedModes = []  
+	if(cmd.heat) { tSupportedModes << "heat" }
+	if(cmd.autoChangeover) { tSupportedModes << "autochangeover" }
+	if(cmd.dryAir) { tSupportedModes << "dryair" }
+	if(cmd.energySaveHeat) { tSupportedModes << "energysaveheat" }
+	if(cmd.away) { tSupportedModes << "away" }
+	if(cmd.off) { tSupportedModes << "off" }    
+	state.supportedModes = tSupportedModes
+	sendEvent(name: "supportedThermostatModes", value: tSupportedModes, displayed: false)
+}
+
+def setHeatingSetpoint(tValue) {
+	def cmds = []
+	def mode = device.currentValue("thermostatMode")
+	def currentMode = modetonumMap[mode]
```

Comment:
`modetonumMap` - Could You camelCase it ?

Reply:
Do you mean to change it to "setHeatingSetPoint" ?
Note that, Setpoint is one word here